runtime/src/main/java/org/corfudb/protocols/wireprotocol/ClientHandshakeHandler.java

```diff
@@ -129,12 +137,16 @@ public void channelRead(ChannelHandlerContext ctx, Object m)
         }
 
         log.info("channelRead: Handshake succeeded. Server Corfu Version: [{}]", corfuVersion);
-        log.debug("channelRead: There are [{}] messages in queue to be flushed.", this.messages.size());
-        // Flush messages in queue
+
+        // Flush messages in queues
         while (!messages.isEmpty()) {
             ctx.writeAndFlush(messages.poll());
         }
 
+        while (!requestMessages.isEmpty()) {
```

Comment:
1. we'd have a race condition here, in a multithreaded environment having a LinkedList. causing race conditions.

2. Could you please replace it with a `BlockingQueue`? 

3. Next step - I don't see the motivation for having a queue here, can we replace it with ConcurrentHashMap? or concurrent set (`ConcurrentHashMap.newKeySet()`)?

4. Since you touched it, could you please also replace `messages` with a concurrent version or a concurrenthashmap? In this class and in serverhandler?

Reply:
Done! Used a `ConcurrentHashMap` as suggested.